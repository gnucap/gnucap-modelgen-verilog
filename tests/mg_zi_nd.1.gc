attach ./modelgen_0.so
attach ../vams/vsine.so

verilog

`modelgen
module test_zi_nd1(p, n);
	(* desc="" *) real dv1;
	(* desc="" *) real dv2;
	(* desc="" *) real dv3;
	(* desc="" *) real dv4;
	(* desc="" *) real dv5;
	electrical p, n;
	analog begin
		dv1 =       zi_nd(V(p) - V(n), '{1}, '{1,2}, .2, .01);
		dv2 = - 2 * zi_nd(V(n) - V(p), '{1}, '{1,2}, .2, .01);
		dv3 = - 3 * zi_nd(-V(p,n),     '{1}, '{1,2}, .2, .01);
		dv4 = - 4 * zi_nd(V(p,n),      '{1}, '{-1,-2}, .2, .01);
		dv5 = - 5 * zi_nd(-V(n,p),    '{-1}, '{1,2}, .2, .01);
	end
endmodule

!make test_zi_nd1.so > /dev/null
attach ./test_zi_nd1.so

verilog
parameter r=1
parameter v=0

test_zi_nd1 #() dut(1,0);
vsine #(.dc(v), .ampl(1)) v1(1, 0);

list

print tran v(1) dv1(dut) dv2(dut) dv3(dut) dv4(dut) dv5(dut)
tran 1 trace=a
status notime
end
