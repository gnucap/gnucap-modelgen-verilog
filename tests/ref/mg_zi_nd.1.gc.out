parameter r=1 
parameter v=0 
test_zi_nd1 #() dut (.p(1),.n(0));
vsine #(.ampl(1),.dc(v)) v1 (.p(1),.n(0));
#Time       v(1)       dv1(dut)   dv2(dut)   dv3(dut)   dv4(dut)   dv5(dut)  
 0.         0.         0.         0.         0.         0.         0.        
 0.01       0.062791   0.         0.         0.         0.         0.        
 0.2        0.95106    0.         0.         0.         0.         0.        
 0.21       0.96858    0.95106    1.9021     2.8532     3.8042     4.7553    
 0.4        0.58779    0.95106    1.9021     2.8532     3.8042     4.7553    
 0.41       0.53583    0.58779    1.1756     1.7634     2.3511     2.9389    
 0.6       -0.58779    0.58779    1.1756     1.7634     2.3511     2.9389    
 0.61      -0.63742   -2.4899    -4.9798    -7.4697    -9.9596    -12.449    
 0.8       -0.95106   -2.4899    -4.9798    -7.4697    -9.9596    -12.449    
 0.81      -0.92978   -2.1266    -4.2533    -6.3799    -8.5065    -10.633    
 1.         14.f      -2.1266    -4.2533    -6.3799    -8.5065    -10.633    
Gnucap   System status
iterations: op=0, dc=0, tran=33, fourier=0, total=44
transient timesteps: accepted=11, rejected=0, total=11
nodes: user=1, subckt=0, model=6, total=7
dctran density=38.8%, ac density=38.8%
